classes/endpoint.php

```diff
@@ -45,16 +45,16 @@ public function register_routes() {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_oembed_response( WP_REST_Request $request ) {
+		if ( 'xml' === $request['format'] ) {
+			add_filter( 'rest_pre_serve_request', array( $this, 'rest_pre_serve_request' ), 10, 4 );
+		}
```

Comment:
You're right, that could lead to problems. Just updated this part and moved it out of the callback. Please let me know how it can be further improved.